ios/Exponent/Versioned/Core/UniversalModules/EXNotifications/EXScopedNotificationCategoriesModule.m

```diff
@@ -22,8 +25,9 @@ - (void)getNotificationCategoriesAsyncWithResolver:(UMPromiseResolveBlock)resolv
 {
   [[UNUserNotificationCenter currentNotificationCenter] getNotificationCategoriesWithCompletionHandler:^(NSSet<UNNotificationCategory *> *categories) {
     NSMutableArray* existingCategories = [NSMutableArray new];
+    NSString *escapedExperienceId = [EXScopedNotificationsUtils escapedString:self->_experienceId];
     for (UNNotificationCategory *category in categories) {
-      if ([category.identifier hasPrefix:self->_experienceId]) {
+      if ([category.identifier hasPrefix:escapedExperienceId]) {
```

Comment:
Instead of doing a prefix match, we can parse the scoped category identifiers into `(scopeKey, unscopedCategoryIdentifier)` pairs and look for matches between `experienceId` and `scopeKey`.

I believe `+[EXScopedNotificationsUtils escapedString:]` then doesn't need to be publicly declared in the headers, which seems right -- other classes should not care about escaping, which is an implementation detail of scoping.